### PR TITLE
Add support for Water Heater Thermex IF series V (pro) Wi-Fi

### PR DESCRIPTION
**Pull request by g470258**

### CODE REVIEW

**Comment by g470258** [comment]
thermex_IF_50_V with quality of 80%. DPS: {"updated_at": 1686045567.5918133, "101": false, "102": 37, "104": 65, "105": "2", "106": 0}
I don't know where to define dp 106. Always with state 0 and never changes.

**Comment by g470258** [comment]
{"updated_at": 1686045567.5918133, "101": false, "102": 37, "104": 65, "105": "2", "106": 0}

File: custom_components/tuya_local/devices/thermex_IF_50_V.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      type: string
+      name: operation_mode
+      mapping:
+        - dps_val: 3
+          constraint: power
+          conditions:
+            - dps_val: false
+              value_redirect: power
+            - dps_val: true
+              value: Economy
```
[the remark this answers is not in the document]
Comment (make-all) [reply custom_components/tuya_local/devices/thermex_IF_50_V.yaml:34]: Based on the implementation of away_mode I just made, the revised mapping should be:
```
 mapping:
          - dps_val: false
            value: "off"
          - dps_val: true
            constraint: work_mode
            conditions:
              - dps_val: "1"
                value: performance
              - dps_val: "2"
                value: electric
                default: true
              - dps_val: "3"
                value: eco
              - dps_val: "4"
                value: away
```

**Review by make-all — CHANGES_REQUESTED** [review]
see comments above

File: custom_components/tuya_local/translations/ru.json
[the remark this answers is not in the document]
Comment (make-all) [reply custom_components/tuya_local/translations/ru.json:1]: I will probably make the tweaks to the device config myself today, so don't worry about separating them.
I just thought the translations could go in straight away, while the config was being worked on.

File: custom_components/tuya_local/devices/thermex_IF_50_V.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (g470258) [reply custom_components/tuya_local/devices/thermex_IF_50_V.yaml:34]: Didn't understand what to rename?
```  dps:
    - id: 101
      type: boolean
      name: operation_mode
      hidden: true
      mapping:
        - dps_val: false
          value: "off"
    - id: 105
      type: string
      name: work_mode
      mapping:
          - dps_val: false
            value: "off"
          - dps_val: true
            constraint: work_mode
            conditions:
              - dps_val: "1"
                value: performance
              - dps_val: "2"
                value: electric
                default: true
              - dps_val: "3"
                value: eco
              - dps_val: "4"
                value: away       
```

**Pull request merged**